src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/MemoryAddressImpl.java

```diff
@@ -119,4 +128,22 @@ public boolean equals(Object that) {
     public String toString() {
         return "MemoryAddress{ region: " + segment + " offset=0x" + Long.toHexString(offset) + " }";
     }
+
+    // helper methods
+
+    public static long addressof(MemoryAddress address) {
+        MemoryAddressImpl addressImpl = (MemoryAddressImpl)address;
+        if (addressImpl.unsafeGetBase() != null) {
+            throw new IllegalStateException("Heap address!");
+        }
+        return addressImpl.unsafeGetOffset();
+    }
+
+    public static MemoryAddress ofLongUnchecked(long value) {
```

Comment:
I believe these two are probably unused?

Reply:
You're right, thanks for catching!